src/MemoryTracing/MemoryTracingUtils.cpp

```diff
@@ -55,75 +49,106 @@ SystemMemoryUsage ParseMemInfo(std::string_view meminfo_content) {
     if (!absl::SimpleAtoi(splits[1], &memory_size_value)) continue;
 
     if (splits[0] == "MemTotal:") {
-      memory_info.set_total_kb(memory_size_value);
+      system_memory_usage->set_total_kb(memory_size_value);
     } else if (splits[0] == "MemFree:") {
-      memory_info.set_free_kb(memory_size_value);
+      system_memory_usage->set_free_kb(memory_size_value);
     } else if (splits[0] == "MemAvailable:") {
-      memory_info.set_available_kb(memory_size_value);
+      system_memory_usage->set_available_kb(memory_size_value);
     } else if (splits[0] == "Buffers:") {
-      memory_info.set_buffers_kb(memory_size_value);
+      system_memory_usage->set_buffers_kb(memory_size_value);
     } else if (splits[0] == "Cached:") {
-      memory_info.set_cached_kb(memory_size_value);
+      system_memory_usage->set_cached_kb(memory_size_value);
     }
   }
+}
 
-  return memory_info;
+void GetValuesFromVmStat(std::string_view vmstat_content, SystemMemoryUsage* system_memory_usage) {
+  if (vmstat_content.empty()) return;
+
+  std::vector<std::string> lines = absl::StrSplit(vmstat_content, '\n');
+  if (lines.empty()) return;
+
+  for (std::string_view line : lines) {
+    // Each line of the /proc/vmstat file consists a single name-value pair, delimited by white
+    // space. In /proc/vmstat, the pgfault and pgmajfault fields report cumulative values.
+    std::vector<std::string> splits = absl::StrSplit(line, ' ', absl::SkipWhitespace{});
+    if (splits.size() < 2) continue;
+
+    int64_t value;
+    if (!absl::SimpleAtoi(splits[1], &value)) continue;
+
+    if (splits[0] == "pgfault") {
+      system_memory_usage->set_pgfault(value);
+    } else if (splits[0] == "pgmajfault") {
+      system_memory_usage->set_pgmajfault(value);
+    }
+  }
 }
 
 ErrorMessageOr<SystemMemoryUsage> GetSystemMemoryUsage() noexcept {
-  uint64_t current_timestamp_ns = orbit_base::CaptureTimestampNs();
-
+  SystemMemoryUsage system_memory_usage;
+  system_memory_usage.set_timestamp_ns(orbit_base::CaptureTimestampNs());
+  system_memory_usage.set_total_kb(kMissingInfo);
```

Comment:
Again, I would prefer to have all of them initialized by default to kMissingInfo

Reply:
As we discussed offline: I will move these value setting lines to the corresponding Get method. If so, we can remove those Reset methods in the unit test.